test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3541 @@
+          ]);
+
+        await this.vault
+          .connect(managerSigner)
+          .commitAndClose([
+            params.asset,
+            params.strikeAsset,
+            params.collateralAsset,
+            secondOption.expiry.toString(),
+            parseEther(params.secondOptionStrike.toString()),
+            this.optionType,
+            params.asset,
+          ]);
+      });
+    });
+
+    describe("#closeShort", () => {
+      time.revertToSnapshotAfterEach(async function () {
+        await depositIntoVault(
+          params.depositAsset,
+          this.vault,
+          this.depositAmount
+        );
+
+        this.oracle = await setupOracle(params.underlyingPricer, ownerSigner);
+
+        if (params.depositAsset === WETH_ADDRESS) {
+          const weth = this.assetContract.connect(counterpartySigner);
+          await weth.deposit({ value: this.premium });
+          await weth.approve(SWAP_ADDRESS, this.premium);
+          return;
+        }
+
+        if (params.mintConfig) {
+          await mintToken(
+            this.assetContract,
+            params.mintConfig.contractOwnerAddress,
+            counterpartySigner,
+            SWAP_ADDRESS,
+            this.premium
+          );
+          return;
+        }
+      });
+
+      it("doesnt do anything when no existing short", async function () {
+        const tx = await this.vault.closeShort();
+        await expect(tx).to.not.emit(this.vault, "CloseShort");
+      });
+
+      it("reverts when closing short before expiry", async function () {
+        await this.vault
+          .connect(managerSigner)
+          .commitAndClose(this.optionTerms);
+
+        await time.increaseTo(
+          (await this.vault.nextOptionReadyAt()).toNumber() + 1
+        );
+
+        await this.vault.connect(managerSigner).rollToNextOption();
+
+        await expect(this.vault.closeShort()).to.be.revertedWith(
+          "Cannot close short before expiry"
+        );
+      });
+
+      it("closes the short after expiry", async function () {
+        await this.vault
+          .connect(managerSigner)
+          .commitAndClose(this.optionTerms);
+
+        await time.increaseTo(
+          (await this.vault.nextOptionReadyAt()).toNumber() + 1
+        );
+
+        await this.vault.connect(managerSigner).rollToNextOption();
+
+        let settlementPrice = this.isPut
+          ? BigNumber.from("7880000000000")
+          : BigNumber.from("158000000000").sub(BigNumber.from("1"));
+
+        await setOpynOracleExpiryPriceYearn(
+          this.asset,
+          this.oracle,
+          settlementPrice,
+          this.collateralPricerSigner,
+          await this.vault.currentOptionExpiry()
+        );
+
+        const closeTx = await this.vault.closeShort();
+
+        assert.isTrue((await this.vault.lockedAmount()).isZero());
+
+        assert.equal(
+          (await this.vault.totalBalance()).toString(),
+          this.depositAmount.sub(1).toString()
```

Comment:
you will notice these .sub(1)'s everywhere.

some rounding weirdness around -1 errors so instead of 90000000000 its 89999999999. believe its got to do with wmul pricePerShare + wdiv pricePerShare (doing these two operations one after other will not get you back to original clearly there will be smallest decimal off by one error). very small amount difference so don't think its a big deal

Reply:
Do you mind adding this comment into the contract function `assetBalance`

Reply:
looks like its actually an issue with yvweth's deposit() - when you wrap your entire eth balance and multiply by pricePerShare it is one wei less than eth balance before wrap. so imprecision comes from imprecise coordination between how much yvweth you get from deposit and pricepershare at that point in time - made comment in rollToNextOption